flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -1364,6 +1365,7 @@ private void closeJob(JobTable.Job job, Exception cause) {
 			jobManagerConnection -> disconnectJobManagerConnection(jobManagerConnection, cause));
 
 		job.close();
+		blobCacheService.getPermanentBlobService().releaseJob(job.getJobId());
```

Comment:
This is correct. I actually wanted to do this as a future follow-up since the PR was already big enough. I'll check how much work it would be to change.

Reply:
It seems to be easily possible. I will add another commit which adds this functionality.